src/main/Video.tsx

```diff
@@ -132,17 +134,29 @@ const VideoPlayer: React.FC<{dataKey: number, url: string, isPrimary: boolean}>
     }
   }
 
-  // Callback for checking whether the video element is ready
-  const onReadyCallback = () => {
-    setReady(true);
-
+  // Tries to get video dimensions from the HTML5 elements until they are not 0,
+  // then updates the store
+  async function updateAspectRatio() {
     if (ref.current && ref.current.getInternalPlayer()) {
       let w = (ref.current.getInternalPlayer() as HTMLVideoElement).videoWidth
       let h = (ref.current.getInternalPlayer() as HTMLVideoElement).videoHeight
+      while (w === 0 || h === 0) {
+        await sleep(1000);
```

Comment:
Would it make sense to make this only a tenth of a second  to be more responsive at startup?

```suggestion
        await sleep(100);
```